osu.Game/Overlays/Profile/Header/BottomHeaderContainer.cs

```diff
@@ -131,7 +132,14 @@ private void updateDisplay(User user)
             tryAddInfo(FontAwesome.Brands.Discord, user.Discord);
             tryAddInfo(FontAwesome.Brands.Skype, user.Skype, @"skype:" + user.Skype + @"?chat");
             tryAddInfo(FontAwesome.Brands.Lastfm, user.Lastfm, $@"https://last.fm/users/{user.Lastfm}");
-            tryAddInfo(FontAwesome.Solid.Link, websiteWithoutProtcol, user.Website);
+            tryAddInfo(FontAwesome.Solid.Link, websiteWithoutProtocol, user.Website);
+
+            // Hide the container to prevent adding unnecessary padding if it has no children other than the NewLineContainer
```

Comment:
i'd prefer `tryAddInfo` didn't add anything in the case of null/empty string, rather than this check.

Reply:
This could be accomplished if `tryAddInfo` was made to return `bool` [as I proposed earlier](https://github.com/ppy/osu/pull/7686#discussion_r373139469), and the addition of the new line was guarded by checking the `anyInfoAdded` flag.

Reply:
that check would still happen before attempting to add second-line info, resulting in the same situation we have on master if no further info was added afterwards.

imo the check has to stay, I could maybe change it to make use of the proposed `anyInfoAdded` instead of the type check, but it would still happen at the end and work virtually the same way with more clutter.

Reply:
I see - the hide is necessary due to the parent fill flow setting vertical spacing.

There's one more benefit to the `anyInfoAdded` variant, however - the children check could be replaced by just checking `anyInfoAdded`, which is probably less surprising than checking children types.

I don't know about clutter, we use similar patterns in other places ([example](https://github.com/ppy/osu/blob/52b66fdc167059ca0336dde474e2ee84073d04b1/osu.Game/Screens/Select/Carousel/CarouselBeatmap.cs#L24-L61)).

Reply:
I've pushed a better (IMO) solution to this, please check it to make sure you agree.

Reply:
I was initially opposed but after re-reading it a few times and considering alternatives it grew on me, so no objections here 👍

Reply:
Seems okay to me as well, except for the spacing difference now:

![image](https://user-images.githubusercontent.com/27722894/73609974-fdb27080-45d2-11ea-885a-d373ea597350.png)

That's due to increased spacing in the combined container, here's how it looks on web (previous version here matched it):

![image](https://user-images.githubusercontent.com/27722894/73610003-4ec26480-45d3-11ea-9f3e-cfed5f1db719.png)

Can this be done 1:1 without splitting the containers to change the newline spacing?

Reply:
After reverting to the 2-container setup and checking, it was actually not the NewLine's fault for adding the extra spacing - even if its not added at all, the spacing is there because `bottomLinkContainer` exists no matter what, which makes the parent `FillFlowContainer` add the specified spacing between its children.

To bring back proper spacing I think it would be best to go back to returning `bool` from `tryAddInfo` and just hide the bottom container if no info was added.

Reply:
I think you could also keep 30a5835, remove spacing on the merged `linkContainer`, and insert *two* newlines if necessary after the first line. Sorta hacky, though. Not sure which is best.

Reply:
That sounds a bit too hacky for me. I decided to revert peppy's changes and do it [the way you proposed](https://github.com/ppy/osu/pull/7686#discussion_r373139469) at the very beginning.

Should be good to go, IMO.

Reply:
I've fixed up a couple of remaining nits in this area, but other than that I think so too.